src/test/java/com/ericsson/ei/jmespath/test/TestIncompleteJmesPathFilter.java

```diff
@@ -0,0 +1,71 @@
+/*
+   Copyright 2018 Ericsson AB.
+   For a full list of individual contributors, please see the commit history.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+package com.ericsson.ei.jmespath.test;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.Test;
+
+import com.ericsson.ei.jmespath.JmesPathInterface;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class TestIncompleteJmesPathFilter {
+    private JmesPathInterface unitUnderTest;
+    private final String aggregatedObjectFilePath = "src/test/resources/AggregatedDocumentInternalCompositionLatest.json";
+
+    @Test
+    public void testFilterObjectWithWholePath() throws IOException {
+        unitUnderTest = new JmesPathInterface();
+        String jsonInput = null;
+        JsonNode expectedResult = null;
+        jsonInput = FileUtils.readFileToString(new File(aggregatedObjectFilePath), "UTF-8");
+        ObjectMapper mapper = new ObjectMapper();
+        expectedResult = mapper.readTree("{\"eventId\":\"33d05e6f-9bd9-4138-83b6-e20cc74680a3\"}");
+        String processRule = "{eventId : incomplete_path_filter(@, 'aggregatedObject.publications[0].eventId')}";
+        JsonNode result = unitUnderTest.runRuleOnEvent(processRule, jsonInput);
+        assertEquals(result, expectedResult);
+    }
+
+    @Test
```

Comment:
add also a test that results in two rows with two different keys

Reply:
Done